index.js

```diff
@@ -43,6 +43,74 @@ exports.Store = Store;
 exports.Cookie = Cookie;
 exports.Session = Session;
 exports.MemoryStore = MemoryStore;
+exports.sessionId = {
+  /**
+   * Set sessionId to cookie.
+   *
+   * @private
+   */
+  set: function (res, name, sessionId, secret, options) {
+    var signed = 's:' + signature.sign(sessionId, secret);
+    var data = cookie.serialize(name, signed, options);
+
+    debug('set-cookie %s', data);
+
+
+    var prev = res.getHeader('set-cookie');
+    var header = [];
+
+    if (prev) {
+      header = header.concat(prev);
+    }
+
+    res.setHeader('set-cookie', header.concat(data))
+  },
+
+  /**
+   * Get the session ID cookie from request.
+   *
+   * @return {string}
+   * @private
+   */
+  get: function (req, name, secrets) {
+    var cookies = cookie.parse(req.headers.cookie || '');
+    var raw = req.cookies && req.cookies[name];
+    var val = req.signedCookies && req.signedCookies[name];
+
+    if (raw || val) {
+      deprecate('cookie should be available in req.headers.cookie');
+    }
+
+    raw = cookies[name] || raw;
+
+    if (!raw) {
+      return val;
+    }
+
+    if (raw.slice(0, 2) !== 's:') {
+      debug('cookie unsigned');
+      return;
+    }
+
+    val = unsigncookie(raw.slice(2), secrets);
+
+    if (val === false) {
+      debug('cookie signature invalid');
+    }
+
+    return val || undefined;
+  },
+
+  /**
+   * Generate a session ID for a new session.
+   *
+   * @return {String}
+   * @private
+   */
+  generate: function () {
```

Comment:
Just reference the old `generateSessionId` function instead of copy and pasting it up here (where it ends up with another level of indentation).